### ISSUE TITLE
Missing `drivers/base/test` directory when compiling

### ISSUE THREAD

**Issue by ardera** [cuu/last_linux-5.4#1]
I'm trying to follow the instructions at https://github.com/clockworkpi/DevTerm/wiki/Create-DevTerm-R01-OS-image-from-scratch#steps-for-compiling-kernel. I want to use mainline gcc though, because it's probably more recent than the allwinner one.

This is what I did:

```console
$ git clone https://github.com/cuu/last_linux-5.4.git
$ cd last_linux-5.4/
$ LANGUAGE=en make CROSS_COMPILE=riscv64-linux-gnu- ARCH=riscv modules dtbs zImage
***
*** Configuration file ".config" not found!
***
*** Please run some configurator (e.g. "make oldconfig" or
*** "make menuconfig" or "make xconfig").
***
Makefile:688: include/config/auto.conf.cmd: No such file or directory
make: *** [Makefile:697: .config] Error 1
```

Since it's apparently missing the config file I paste file https://github.com/clockworkpi/DevTerm/blob/main/Code/patch/d1/config as `.config`:
```console
$ LANGUAGE=en wget -O .config https://raw.githubusercontent.com/clockworkpi/DevTerm/main/Code/patch/d1/config
--2022-06-09 15:59:44--  https://raw.githubusercontent.com/clockworkpi/DevTerm/main/Code/patch/d1/config
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 2606:50c0:8003::154, 2606:50c0:8000::154, 2606:50c0:8001::154, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|2606:50c0:8003::154|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 123296 (120K) [text/plain]
Saving to: ‘.config’

.config                    100%[======================================>] 120,41K  --.-KB/s    in 0,03s   

2022-06-09 15:59:44 (3,79 MB/s) - ‘.config’ saved [123296/123296]

$ LANGUAGE=en make CROSS_COMPILE=riscv64-linux-gnu- ARCH=riscv modules dtbs zImage
  HOSTCC  scripts/basic/fixdep
  HOSTCC  scripts/kconfig/conf.o
  HOSTCC  scripts/kconfig/confdata.o
  HOSTCC  scripts/kconfig/expr.o
  LEX     scripts/kconfig/lexer.lex.c
  YACC    scripts/kconfig/parser.tab.[ch]
  HOSTCC  scripts/kconfig/lexer.lex.o
  HOSTCC  scripts/kconfig/parser.tab.o
  HOSTCC  scripts/kconfig/preprocess.o
  HOSTCC  scripts/kconfig/symbol.o
  HOSTLD  scripts/kconfig/conf
scripts/kconfig/conf  --syncconfig Kconfig
drivers/base/Kconfig:159: can't open file "drivers/base/test/Kconfig"
make[2]: *** [scripts/kconfig/Makefile:73: syncconfig] Error 1
make[1]: *** [Makefile:590: syncconfig] Error 2
Makefile:688: include/config/auto.conf.cmd: No such file or directory
make: *** [Makefile:706: include/config/auto.conf.cmd] Error 2
$
```

It's apparently missing some test directories that have been deleted. Any idea what's going on?

**Comment by grant-olson** [comment]
The issue is that there is an addition from allwinner to the global `.gitignore` that causes all test directories to be ignored. Since this did a new check in starting at commit one they didn't get included.

@cuu the following diff should reveal the missing directories so that you can add them to this repo:

```
diff --git a/.gitignore b/.gitignore
index 3a1f67d11..885cfb5b6 100644
--- a/.gitignore
+++ b/.gitignore
@@ -156,7 +156,6 @@ output/
 arch/arm64/boot/dts/sunxi/board.dts
 arch/arm/boot/dts/board.dts
 #arch/riscv/boot/dts/sunxi/board.dts
-test/
 test_guu/
 
 #lto lds
```

Let me know if you would like a proper pull request.

**Comment by ardera** [comment]
fixed by a93ac38b8afb526c534003619ca9f0a7a0be0e06
